Bugfix/table cells interactive focusable

Mahmoud-zino opened the pull request:
## Linked Issue

Closes #2168

## Description

table cells are focusable only in interactive mode.

## Changsets

Instructions: Changesets automate our changelog. If you modify files in `/packages/skeleton`, run `pnpm changeset` in the root of the monorepo, follow the prompts, then commit the markdown file. Changes that add features should be `minor` while chores and bugfixes should be `patch`. Please prefix the changeset message with `feat:`, `bugfix:` or `chore:`.

## Checklist

Please read and apply all [contribution requirements](https://www.skeleton.dev/docs/contributing).

- [x] This PR targets the `dev` branch (NEVER `master`)
- [x] Documentation reflects all relevant changes
- [x] Branch is prefixed with: `docs/`, `feat/`, `chore/`, `bugfix/`
- [x] Ensure Svelte and Typescript linting is current - run `pnpm check`
- [x] Ensure Prettier linting is current - run `pnpm format`
- [x] All test cases are passing - run `pnpm test`
- [x] Includes a changeset (if relevant; see above)

endigo9740 commented:
@Mahmoud-zino I think you're on the right track, but we need to account for a11y here too. The WAI APG define two types of interactive table-like elements:

1. [Table Pattern](https://www.w3.org/WAI/ARIA/apg/patterns/table/) - like a presentation HTML table, but not interactive
2. [Grid Pattern](https://www.w3.org/WAI/ARIA/apg/patterns/grid/) - these are like a table, but interactive

While changing the `tabindex` is likely correct, it doesn't quite go far enough. I think we'll likely also need to modify following two roles:

- `role="grid"` -> `role="table"`
- `role="gridcell"` -> `role="cell"`

I'm also questioning if we should introduce `row` and `columnheader` roles, or if our use of a native HTML table is semantic enough as is.

I originally tested stuff like this using this Chrome Extension. Feel free to use it to double check:
https://chrome.google.com/webstore/detail/axe-devtools-web-accessib/lhdoppojpmngadmnindnejefpokejbdd

endigo9740 commented:
@ryceg I know you're fairly knowledgable with the a11y stuff, so I'd welcome your input here as well

Mahmoud-zino commented:
@endigo9740 I think it is better to separate the components to grid / table and handling all ARIA rules for each one of them instead of trying to handle both in the same component

endigo9740 commented:
@Mahmoud-zino if it's only the couple things mentioned I don't think separate components would be needed. Honestly most of the ARIA stuff overlaps as far as I can tell. Anything further abstraction would just make these harder to maintain over time.

endigo9740 commented:
Hey @Mahmoud-zino, I'm doing PR reviews on Sundays now. Just checking in on this. As far as I can tell we should just need to modify the tools per the requirements I mentioned above. I'll aim to check in on this again Tuesday before the new release. Thanks!

Mahmoud-zino commented:
@endigo9740 I implemented the roles you mentioned and gone through the ARIA articles and implemented those that are not redundant roles.
Please feel free to suggest changes if needed.
